packages/common/src/api/tan-query/jupiter/useSwapTokens.ts

```diff
@@ -0,0 +1,375 @@
+              outputAmount: quoteResult.outputAmount
+            }
+          }
+        }
+
+        // --- 3b. Determine Jupiter's Destination Token Account ---
+        let preferredJupiterDestination: string | undefined
+        let outputUserBankAddress: PublicKey | undefined
+
+        const isOutputUserBankManaged = !!(
+          outputTokenConfig &&
+          ethAddress &&
+          outputMintUiAddress.toUpperCase() !== 'SOL'
+        )
+        if (isOutputUserBankManaged) {
+          try {
+            outputUserBankAddress =
+              await sdk.services.claimableTokensClient.deriveUserBank({
+                ethWallet: ethAddress!,
+                mint: outputTokenConfig!.claimableTokenMint
+              })
+            preferredJupiterDestination = outputUserBankAddress.toBase58()
+          } catch (error: any) {
+            reportToSentry({
+              name: 'JupiterSwapOutputUserBankDerivationError',
+              error,
+              feature: Feature.TanQuery,
+              additionalInfo: { params }
+            })
+          }
+        }
+
+        // --- 3c. Get Jupiter Swap Instructions ---
+        const {
+          tokenLedgerInstruction,
+          swapInstruction,
+          addressLookupTableAddresses
+        } = await jupiterInstance.swapInstructionsPost({
+          swapRequest: {
+            quoteResponse: quoteResult.quote,
+            userPublicKey: userPublicKey.toBase58(),
+            destinationTokenAccount: preferredJupiterDestination,
+            wrapAndUnwrapSol: wrapUnwrapSol,
+            useSharedAccounts: true
+          }
+        })
+
+        const jupiterInstructions = convertJupiterInstructions([
+          tokenLedgerInstruction,
+          swapInstruction
+        ])
+
+        updateJupiterAtaCreationFeePayer(
+          jupiterInstructions,
+          userPublicKey,
+          feePayer
+        )
+        instructions.push(...jupiterInstructions)
+
+        // --- 3d. Identify Actual Jupiter Destination & Potential Jupiter-Created Temporary ATA ---
+        const actualJupiterDestination =
+          findActualJupiterDestination(jupiterInstructions)
+        const jupiterTemporarySetupAta = outputTokenConfig
+          ? findJupiterTemporarySetupAta(
+              jupiterInstructions,
+              outputTokenConfig.mintAddress,
+              actualJupiterDestination
+            )
+          : undefined
+
+        // --- 3e. Handle Output Token (if user bank managed and not directly deposited by Jupiter) ---
+        let wasOutputDepositedToUserBank = false
+        if (
+          outputUserBankAddress &&
+          actualJupiterDestination?.equals(outputUserBankAddress)
+        ) {
+          wasOutputDepositedToUserBank = true
+        }
+
+        const needsManualTransferToOutputUserBank =
+          isOutputUserBankManaged &&
+          actualJupiterDestination &&
+          !wasOutputDepositedToUserBank
+
+        if (needsManualTransferToOutputUserBank) {
```

Comment:
do we ever need this? does jupiter sometimes not send to the destinationTokenAccount?